### DISCUSSION TITLE
Fulfillment Inbound API v2024-03-20.

### DISCUSSION THREAD

**Discussion by IncredibleX** [abuzuhri/Amazon-SP-API-CSharp#744] [General]
Hi, i got the following mail. Is there anything planned?

---
We have deprecated 10 operations in Fulfillment Inbound API v0 with a removal date of December 20, 2024:

• createInboundShipmentPlan
• updateInboundShipment
• createInboundShipment
• getPreorderInfo
• confirmPreorder
• getTransportDetails
• putTransportDetails
• voidTransport
• estimateTransport
• confirmTransport

Thanx,

Tobias

**Comment by Cw733** [comment]
@abuzuhri  @J-W-Chan @RenzoF 

Can you guys give us any idea if you are able to review and merge pull requests, or is there some way we can help?  Right now there is a PR #730 from @kevinvenclovas that addresses the the new Inbound Fulfillment, although it may not follow the same coding standards as the rest of the library, and has some additional changes outside of just the Inbound Fulfillment.

Is that something you guys can look at merging, or do you need help due to time?  I'd like to see this project continue and I'd be willing to dedicate some time to it, but it doesn't make sense for several of us to implement the new APIs individually and then only one of them get merged into the original project.

Can you guys let us know your ideas about keeping the project up to date, and how we can help?

Thanks,
Casey

**Comment by abuzuhri** [comment]
Usually we merge fast but this branch so big and not same our sliced style
and will need big time to review and I don’t have that time now .

If you can help on adding new api but with same code style will be nice and
will merge fast for you

**Reply by Cw733** [reply]
Thanks Tareq, I thought that might be the case.  I'll do my best to work on it as we will need it before the Dec. 30 2024 deadline.

Would you like a new namespace created that references the new API version, e.g. "FulfillmentInbound2024-03-20," and leave the original V0 version in place?  This seems to be the pattern you guys have followed thus far, and I think it makes the most sense, but just want to make sure?

Thanks,
Casey

**Comment by kevinvenclovas** [comment]
Hey,

yes, I used the OpenAPI service to generate the API models. All other code follows the same coding style. Feel free to add the models directly to the project and remove the OpenAPI service.

**Reply by abuzuhri** [reply]
There is many problem came with code generated from NSwagCSharp , the amazon API not well organize and best solution most of developers decide not use it https://github.com/abuzuhri/Amazon-SP-API-CSharp/discussions/528

**Reply by abuzuhri** [reply]
Also i notes that you remove old API , that not our standard work because maybe some one still usd the old API we don't want to damage his code after upgrade

**Reply by abuzuhri** [reply]
@kevinvenclovas is it possible to resubmit it without NSwagCSharp and with new keep old API ?

**Reply by kevinvenclovas** [reply]
@abuzuhri  Check the new pull request :)

**Reply by Cw733** [reply]
Thanks @kevinvenclovas.  If there is anything else I can do to help with this one let me know @abuzuhri.  At first glance this new PR looks good to me.

**Reply by abuzuhri** [reply]
@kevinvenclovas  Thanks for your work , please test v1.7.21 now and also add sample in readme file and add new sample file

**Comment by smargoli2** [comment]
Question about this: The new api does not support the getShipments endpoint where we can get shipments within a date range without shipment ids. This endpoint has not been deprecated as part of the migration, so we can potentially still use it. What do you recommend we do in this situation? Will this endpoint be deprecated eventually as well?
